Sources/BSPLogging/BSPLogging.swift

```diff
@@ -0,0 +1,190 @@
+    ) throws {
+        self.logLevel = logLevel
+        self.metadata = metadata
+        
+        self.label = label
+        self.fileName = fileName
+        self.fileURL = fileDirectory.appending(component: fileName)
+        self.dateFormatter = DateFormatter()
+        self.dateFormatter.dateFormat = "yyyy-MM-dd HH:mm:ss.SSS"
+        
+        if !FileManager.default.fileExists(atPath: fileURL.path()) {
+            FileManager.default.createFile(atPath: fileURL.path(), contents: nil, attributes: nil)
+        } else if truncate {
+            // Truncate existing file
+            try Data().write(to: fileURL)
+        }
+        
+        self.fileHandle = try FileHandleWrapper(fileURL: fileURL)
+    }
+
+    public func log(
+        level: SwiftLogger.Level,
+        message: SwiftLogger.Message,
+        metadata: SwiftLogger.Metadata?,
+        source: String,
+        file: String,
+        function: String,
+        line: UInt
+    ) {
+        let timestamp = self.dateFormatter.string(from: Date())
+        let logEntry = "[\(timestamp)] [\(label)] [\(level.rawValue)] \(message)\n"
+        self.writeToFile(logEntry)
+    }
+    
+    private func writeToFile(_ logEntry: String) {
+        if let data = logEntry.data(using: .utf8) {
+            try? self.fileHandle.write(contentsOf: data)
+        }
+    }
+}
+
+public struct OSLogHandler: LogHandler, Sendable {
+
+    public var metadata: Logging.Logger.Metadata = .init()
+    public subscript(metadataKey metadataKey: String) -> Logging.Logger.Metadata.Value? {
+        get {
+            self.metadata[metadataKey]
+        }
+        set(newValue) {
+            self.metadata[metadataKey] = newValue
+        }
+    }
+    
+    public var logLevel: SwiftLogger.Level = .info
+    
+    public let logger: OSLogger
+    
+    public init(
+        logLevel: SwiftLogger.Level = .info,
+        subsystem: String = "sourcekit-bazel-bsp",
+        category: String = #fileID
+    ) {
+        self.logLevel = logLevel
+        self.logger = OSLogger(subsystem: subsystem, category: category)
+    }
+    
+    public func log(
+        level: SwiftLogger.Level,
+        message: SwiftLogger.Message,
+        metadata: SwiftLogger.Metadata?,
+        source: String,
+        file: String,
+        function: String,
+        line: UInt
+    ) {
+        switch level {
+        case .info:
+            self.logger.info("\(message, privacy: .public)")
+        case .debug:
+            self.logger.debug("\(message, privacy: .public)")
+        case .error:
+            self.logger.error("\(message, privacy: .public)")
+        case .critical:
+            self.logger.critical("\(message, privacy: .public)")
+        case .notice:
+            self.logger.notice("\(message, privacy: .public)")
+        case .warning:
+            self.logger.warning("\(message, privacy: .public)")
+        case .trace:
+            self.logger.trace("\(message, privacy: .public)")
+        }
+    }
```

Comment:
I think you need to use OSLogMessage/StaticStrings here, otherwise this removes the privacy benefits of OSLog. The code today is all public but one of my refactors right now will be to remove them and have "proper" os log usage.